File: Day 28 : RegEx, Patterns, and Intro to Databases/Solution.py
import re

if __name__ == '__main__':
    N = int(input())
    
    final = []
    
    for N_itr in range(N):
        first_multiple_input = input().split()

        firstName = first_multiple_input[0]
        emailID = first_multiple_input[1]
        
        if re.search(".+@gmail\.com$", emailID):
            final.append(firstName)
    
    final.sort()
    
    for i in final:
        print(i)

File: Day 28 : RegEx, Patterns, and Intro to Databases/Solution.rb
N = gets.strip.to_i
names = Array.new

N.times do
  first_multiple_input = gets.rstrip.split

  firstName = first_multiple_input[0]

  emailID = first_multiple_input[1]
  
  regex = /(\@gmail\.com)$/
  
  if regex.match(emailID)
    names.push firstName
  end 
end

puts names.sort
